Libs/MRML/Core/vtkMRMLAbstractLayoutNode.h

```diff
@@ -0,0 +1,61 @@
+/*==============================================================================
+
+  Copyright (c) Laboratory for Percutaneous Surgery (PerkLab)
+  Queen's University, Kingston, ON, Canada. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Csaba Pinter, PerkLab, Queen's University
+  and was supported through CANARIE�s Research Software Program, and Cancer
+  Care Ontario.
+
+==============================================================================*/
+
+#ifndef __vtkMRMLAbstractLayoutNode_h
+#define __vtkMRMLAbstractLayoutNode_h
+
+// MRML includes
+#include "vtkMRMLNode.h"
+
+class vtkXMLDataElement;
+
+/// \brief Node that describes the view layout of the application.
+///
+/// When the scene is closing (vtkMRMLScene::Clear), the view arrangement is
+/// set to none due to the Copy() call on an empty node.
+class VTK_MRML_EXPORT vtkMRMLAbstractLayoutNode : public vtkMRMLNode
+{
+public:
+  vtkTypeMacro(vtkMRMLAbstractLayoutNode,vtkMRMLNode);
+  void PrintSelf(ostream& os, vtkIndent indent) VTK_OVERRIDE;
+
+  /// Set node attributes
+  virtual void ReadXMLAttributes( const char** atts) VTK_OVERRIDE;
+
+  /// Write this node's information to a MRML file in XML format.
+  virtual void WriteXML(ostream& of, int indent) VTK_OVERRIDE;
+
+  /// Copy the node's attributes to this object
+  virtual void Copy(vtkMRMLNode *node) VTK_OVERRIDE;
+
+protected:
+  vtkMRMLAbstractLayoutNode();
+  ~vtkMRMLAbstractLayoutNode();
+
+  vtkMRMLAbstractLayoutNode(const vtkMRMLAbstractLayoutNode&);
+  void operator=(const vtkMRMLAbstractLayoutNode&);
+
+  //TODO: Add features to handle heads-up display. It is unlikely that the VR layouts can use
```

Comment:
I would not say it is a TODO. You can just describe (preferably in the class documentation, so that it shows up in doxygen) what are the intended roles and responsibilities of this class.

Reply:
Right, this belongs in vtkMRMLVRLayoutNode anyway.